src/main/java/com/checkmarx/flow/service/FlowService.java

```diff
@@ -37,10 +37,13 @@ private void runScanRequest(ScanRequest scanRequest, List<VulnerabilityScanner>
 
         scanners.forEach(scanner -> {
             ScanResults scanResults = scanner.scan(scanRequest);
-            combinedResults.mergeWith(scanResults);
+            if(scanResults!=null) {
+                combinedResults.mergeWith(scanResults);
+            }
         });
-
-        resultsService.publishCombinedResults(scanRequest, combinedResults);
+        if(combinedResults.getProjectId()!=null) {
```

Comment:
combinedResults.getProjectId() only refers to SAST project ID. If we use another vulnerability scanner (e.g. SCA without SAST), then combinedResults.getProjectId() will return null. As a result, SCA results will never be published.

Reply:
New condition : if(combinedResults.getSastScanId()!=null || combinedResults.getScaResults()!=null)

Changed the condition to check for null values of SastScanId or ScaResults. The intention here is to avoid further processing (and exceptions) when combinedResults(Type-ScanResults) is completely null. Whenever any exceptions are thrown from the scan() method, it results into scanResults being null. And this gives more exceptions as further processing occurs. The new condition is expected to work for Sast Scans as well Osa Scans.

In case, I missed something, kindly let me know.

Reply:
Also, I installed and started using the SonarLint plugin. Thanks for letting me know.